stacks_queues/queue.py

```diff
@@ -15,47 +15,71 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == INITIAL_QUEUE_SIZE:
+            raise QueueFullException
+        self.size += 1
+        if self.rear == (INITIAL_QUEUE_SIZE - 1):
+            self.rear = 0
+        else:
+            self.rear += 1
+        if self.front == -1:
+            self.front = self.rear
+        self.store[self.rear] = element
 
     def dequeue(self):
```

Comment:
✨ It might also be nice to check if the queue becomes empty after dequeuing the element and point our `front` and `rear` pointers back to `-1` if this is the case.